Menu.md

```diff
@@ -0,0 +1,413 @@
+# RFC Menu / Navigation
+
+## Summary
+
+Today the main pain-points of using our `menu` block along with the other blocks exported by `vtex.menu` are:
+
+- Poor performance, due to a very high number of blocks;
+- Our `menu`is not _interchangeable_, which causes problems such as having to implement **four** distinct `menu`s with the same blocks so that it gets rendered on `header`, `footer`, `mobile` and `desktop`.
+
+The introduction of `list-context` components and the added capability of the `menu-item` block to receive a _list_ of blocks meant we could reduce the number of rendered blocks and thus improve performance, but this solution was limited, since lists could not work with `submenu`s.
+
+In this RFC we propose a new way of dealing with navigation blocks, mainly the way users create menus throughout their stores. We are addressing all of the problems presented above while also keeping in mind the developments being made to create a new CMS for VTEX IO stores, where users will be able to edit this new Menu structures.
+
+## Basic example
+
+This is a basic example of how a new Menu implementation would look like and what the users would need to code. Don't worry if this example does not make sense to you, we'll go deeper into the technical details 
+
+The reference for this example is the current menu implemented by Al's Sporting Goods store:
+
+![als-menu-mens](https://user-images.githubusercontent.com/27777263/77101924-d682f780-69f6-11ea-8510-b4ede5a7692b.png)
+
+Let's focus on just the `Men's` submenu, as shown in the image. To achieve this menu layout, users would write something like this into a `navigation.json`file:
+
+```jsonc
+{
+  "main-navigation": {
+    "id": "39214",
+    "title": "Main Navigation",
+    "items": [
+      {
+        "id": "41201",
+        "label": "Men's",
+        "link": "#",
+        // The value for "child" should be the ID
+        // of another navigation.
+        "child": "39215"
```

Comment:
We should think of a better name than "child"

Reply:
I like child, a more clear name would be `childNavigation`, but I think is not needed

Reply:
I also think `child` is OK, but I agree we could think of something a bit clearer. `childNavigation` is more verbose, but easier to understand.

How about `subNavigation`? Since this navigation will be used in `submenu`s

Reply:
subNavigation is good!